src/shacl2code/lang/templates/python.j2

```diff
@@ -1713,7 +1732,7 @@ class Encoder(ABC):
 
         Each item of the list will be added using `write_list_item`
         """
-        pass
+        yield self
```

Comment:
Since these are abstract, I think they can raise NotImplementedError() which is I think is the better way to deal with this

Reply:
Agreed. I have change this to other abstract methods as well. Thank you.